tests/Http/PostTest.php

```diff
@@ -1349,6 +1359,29 @@ public function testUpdatingAPhotoPost()
         $this->assertEquals($signup->fresh()->quantity, 8);
     }
 
+    /**
+     * Test that updating a referral post sends an event to Customer.io.
+     *
+     * PATCH /api/v3/posts/:id
+     * @return void
+     */
+    public function testUpdatingAReferralPost()
+    {
+        $post = factory(Post::class)->create([
+            'referrer_user_id' => $this->faker->northstar_id,
+            'type' => 'voter-reg',
+            'status' => 'step-1',
+        ]);
+
+        $this->mock(CustomerIo::class)
+          ->shouldReceive('trackEvent')
+          ->with($post->referrer_user_id, 'referral_post_updated', $post->getReferralPostEventPayload());
```

Comment:
any reason we don't assert this above in the post created test?

Reply:
Yeah, this one gets to me - I wish we could. It's because we don't have a `post->id` set -- so the test would fail. The reason I thought it'd be wise to include the post id in the Customer.io event was more for debugging, although I guess we'd need it anyway should we ever link a referrer to a referral post permalink (although seems really rare that we'd do so on post creation, cos it'd be pending in most cases, but you never know)